task/06-conditions-n-loops-tasks.js

```diff
@@ -166,7 +180,17 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+    let map = new Map();
+    str.split('').map(function(item) {
+        if (!map.get(item)) {
+            map.set(item, 0);
+        }
+        map.set(item, map.get(item)+1);
+    })
+
+    const isEqualsOne = (element) => element == 1;
+    let nonRepeatedChar = Array.from(map.keys())[Array.from(map.values()).findIndex(isEqualsOne)];
+    return nonRepeatedChar ? nonRepeatedChar : null;
```

Comment:
ternary operator is redundant here
 Please, see https://javascript.info/logical-operators
we can return just nonRepeatedChar with the same result